src/MovieApp.java

```diff
@@ -0,0 +1,211 @@
+                    break;
+                case 1:
+                    inputMovies();
+                    break;
+                case 2:
+                    displayMovies(movies);
+                    break;
+                case 3:
+                    addMovies();
+                    break;
+                case 4:
+                    findStatistics();
+                    break;
+                case 5:
+                    searchMovie();
+                    break;
+                case 6:
+                    updateRating();
+                    break;
+                case 7:
+                    deleteMovie();
+                    break;
+                case 8:
+                    sortMovie();
+                    break;
+                default:
+                    System.out.println("Invalid input");
+                    break;
+
+            }
+            if (!isRun) break;
+        }
+
+    }
+
+    public static void showMenu() {
+        System.out.println("""
+                Press 1: to input movies
+                Press 2: to display movies
+                Press 3: to add movies
+                Press 4: to find statistics
+                Press 5: to search movie
+                Press 6: to update ratings
+                Press 7: to delete movies
+                Press 8: to sort movies
+                Press 0: to exit
+                """
+        );
+    }
+
+    private void inputMovies() {
+        movies = new Movie[3];
+        for (int i = 0; i < movies.length; i++) {
+            System.out.println("Enter the movie name: ");
+            String name = new Scanner(System.in).nextLine();
+            System.out.println("Enter the movie rating: ");
+            double rating = new Scanner(System.in).nextDouble();
+            movies[i] = new Movie(name, rating);
+        }
+        System.out.println("Added successfully!!");
+    }
+
+    private void displayMovies(Movie[] movies) {
+        for (Movie movie : movies) {
+            System.out.println(movie.getName() + ": " + movie.getRating());
+        }
+    }
+
+    private void addMovies() {
+        System.out.println("Enter the number of movies that you want to add: ");
+        int num = new Scanner(System.in).nextInt();
+        int length = movies.length;
+        Movie[] newMovies = new Movie[num + length];
+        System.arraycopy(movies, 0, newMovies, 0, movies.length);
+
+        for (int i = 0; i < num; i++) {
+            System.out.println("Enter the movie name " + (i + 1) + ": ");
+            String name = new Scanner(System.in).nextLine();
+            System.out.println("Enter the movie rating "+(i+1)+": ");
+            double rating = new Scanner(System.in).nextDouble();
+            newMovies[length + i] = new Movie(name, rating);
+        }
+        movies = newMovies;
+        System.out.println("Added successfully!!");
+    }
+
+    private void findStatistics() {
+        double sum = 0.0;
+        Movie minimum = movies[0];
```

Comment:
you can store just minRating instead Movie itself. as you see you just use rating field of the Movie object, then why we need to store Object